src/Compilers/CSharp/Portable/Syntax/NullableDirectiveMap.cs

```diff
@@ -56,20 +70,28 @@ private NullableDirectiveMap(ImmutableArray<(int Position, bool? State)> directi
             }
 
             // Generated files have an initial nullable context that is "disabled"
-            bool? state = (_isGeneratedCode) ? false : (bool?)null;
+            var directive = _isGeneratedCode
+                ? new NullableDirective(position, false, false)
+                : new NullableDirective(position, null, null);
+
             if (index >= 0)
             {
                 Debug.Assert(_directives[index].Position <= position);
                 Debug.Assert(index == _directives.Length - 1 || position < _directives[index + 1].Position);
-                state = _directives[index].State;
+                directive = _directives[index];
             }
 
-            return state;
+            return directive;
         }
 
-        private static ImmutableArray<(int Position, bool? State)> GetDirectives(SyntaxTree tree)
+        private static ImmutableArray<NullableDirective> GetDirectives(SyntaxTree tree, bool isGeneratedCode)
         {
-            var builder = ArrayBuilder<(int Position, bool? State)>.GetInstance();
+            // Generated files have an initial nullable context that is "disabled"
+            var previousDirective = isGeneratedCode
```

Comment:
>var previousDirective = isGeneratedCode [](start = 12, length = 39)

nit: consider factoring this and similar snippet above (line 73) out to a method `GetDirectiveForFileStart(bool isGeneratedCode)`

Reply:
Marking this closed because I assume your approval includes approval for this change

---
In reply to: [295073254](https://github.com/dotnet/roslyn/pull/36464#discussion_r295073254) [](ancestors = 295073254)